sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3650,12 +3652,24 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       }
     }
 
-    val functionIdentifier = visitMultipartIdentifier(ctx.multipartIdentifier)
-    CreateFunctionStatement(
-      functionIdentifier,
+    val nameParts = visitMultipartIdentifier(ctx.multipartIdentifier)
+    val isTemp = ctx.TEMPORARY != null
+    val func: LogicalPlan = if (isTemp) {
+      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+      // temp func doesn't belong to any catalog and we shouldn't resolve catalog in the name.
+      if (nameParts.length > 2) {
+        throw new AnalysisException(s"Unsupported function name '${nameParts.quoted}'")
+      }
+      ResolvedFunc(nameParts.asIdentifier)
+    } else {
+      UnresolvedFunc(nameParts)
+    }
+
+    CreateFunction(
+      func,
```

Comment:
I'm not sure how to deal with CREATE commands yet. For now, CREATE TABLE and VIEW still use statement plans. We don't need to do lookup for CREATE commands, so `UnresolvedFunnc` looks weird here.

Can we keep it unchanged?

Reply:
Ok. Will revert this change.